### ISSUE TITLE
Assertion Error during FeatureExtractor Training

### ISSUE THREAD

**Issue by roray02** [jbohnslav/deepethogram#45]
Hello,

We've tried going through the steps as mentioned in the workflow model, but seem to be running into various issues when training the feature extractor. 

In one instance, the FeatureExtractor will say that it has finished training, despite running into this assertion error:
![image](https://user-images.githubusercontent.com/69780075/112229239-fa2fdd80-8c08-11eb-9a4b-4eb4a1c34d87.png)

It will also say that there is only 1 finalized label, despite 4 different labels being utilized.

This was the result after we attempted inference using the FeatureExtractor, and the associated assertion error that came with it.
![image](https://user-images.githubusercontent.com/69780075/112229411-4ed35880-8c09-11eb-9081-2034bd20c479.png)

If this is due to some issue in our config.yaml file, how should we go about fixing it, and if not, what might be the root cause of these issues?

Thanks for any help!

**Comment by roray02** [comment]
Yes, we clicked the finalize labels button after fully labeling a video with 4 different labels.

**Comment by jbohnslav** [comment]
Can you try just deleting the `split.yaml` file in your model directory?

**Comment by roray02** [comment]
The split.yaml file doesn't actually appear after training the flow generator in our case. Only the config.yaml, hydra.yaml, and overrides.yaml files appear within the folder.

![image](https://user-images.githubusercontent.com/69780075/112362680-12583900-8cab-11eb-80e2-8cb123bd648f.png)

Would we need to manually add and configure this file in the directory? I was under the assumption that running the flow generator training would automatically generate this file with the split between the train and validation sets. If that's the case, are there any other additions we would need to make to the file structure in order to get things running smoothly? I noticed that the "proper" file structure for the generator training had other files that didn't seem to be generated in our directory when we had run our flow generator training.

![image](https://user-images.githubusercontent.com/69780075/112364643-23a24500-8cad-11eb-9c56-4d302b0d4992.png)

Thanks!

**Comment by jbohnslav** [comment]
Hmm, this is because you're looking in the pretrained models directory it looks like. If you go to the `models` directory, is there anything besides the `pretrained_models` directory? You should see the flow generators there, as well as the `split.yaml` file.

**Comment by roray02** [comment]
The models directory only contains the pretrained_models folder.
![image](https://user-images.githubusercontent.com/69780075/112369053-281d2c80-8cb2-11eb-886b-546def0bec66.png)

Should I get rid of the pretrained_models folder, and directly keep everything in `models`, while changing the path in the config.yaml file?

**Comment by jbohnslav** [comment]
Hmm, this is weird. It looks like no flow_generator models actually trained. Did you train one successfully? What is the `project/model_path` in your `project_config.yaml`?

**Comment by roray02** [comment]
No, I don't believe we were ever able to train one successfully, despite the GUI indicating to us after every instance of training that `training finished`. The `project/model_path` in `project_config.yaml` is just the file path to the models directory, `C:\Users\Ray\Documents\Deep_Ethogram_Trials\try1_deepethogram/models` (we had previously just used `models` as a file path as well, with no luck). For transparency, I am attaching the `project_config.yaml` file to this comment as well in case there's something I'm missing.

```
augs:
  LR: 0.5
  UD: 0.5
  brightness: 0.25
  contrast: 0.1
  crop_size: null
  degrees: 10
  grayscale: 0.0
  normalization:
    N: 1243652096
    mean:
    - 0.5848328077979216
    - 0.5744588576800879
    - 0.5957053103914569
    std:
    - 0.3090533559012072
    - 0.29111276540011655
    - 0.2988417887585355
  pad: null
  random_resize: false
  resize: null
compute:
  batch_size: 32
  distributed: false
  gpu_id: 0
  num_workers: 8
project:
  class_names:
  - background
  - facegroom
  - bodygroom
  - explore
  - rear
  config_file: C:/Users/Ray/Documents/Deep_Ethogram_Trials/try1_deepethogram\project_config.yaml
  data_path: DATA
  labeler: null
  model_path: C:\Users\Ray\Documents\Deep_Ethogram_Trials\try1_deepethogram/models
  name: try1
  path: C:/Users/Ray/Documents/Deep_Ethogram_Trials/try1_deepethogram
sequence:
  filter_length: 15
split:
  file: null
  reload: true
train:
  loss_weight_exp: 0.5
```

**Comment by jbohnslav** [comment]
This all looks fine to me. I'm working on a major update that will hopefully improve this process. Can you try training a feature extractor model from the GUI? It should make a run directory in the `models` folder. In any case, I'll be able to see what's going on due to the error messages on the command prompt.

**Comment by roray02** [comment]
A feature_extractor_train directory was created upon running the training of a feature extractor model, but it doesn't seem to have the `split.yaml` file either.

![image](https://user-images.githubusercontent.com/69780075/112541487-d34ce500-8d89-11eb-8d93-bf5d8c607048.png)
![image](https://user-images.githubusercontent.com/69780075/112541459-c7f9b980-8d89-11eb-9afb-ab268097e09b.png)

This was the command prompt error message:
![image](https://user-images.githubusercontent.com/69780075/112541650-055e4700-8d8a-11eb-80a1-f75d44c1dfbf.png)

**Comment by roray02** [comment]
After fixing some of the assertion error issues, it seems that the training is at least starting, but now does not actually make any progress and doesn't have a finite finish time. The 'split.yaml' file is also now being generated after fixing the assertion errors. I was also wondering if there was a specific date when the major update to DEG that you referred to above was going to be released.

![image](https://user-images.githubusercontent.com/69780075/112898584-02779500-90af-11eb-9813-bc65041ae24b.png)

**Comment by jbohnslav** [comment]
Hi, please update DeepEthogram to the new version [described here](https://github.com/jbohnslav/deepethogram/blob/master/docs/beta.md). If you're still having issues, you can reopen this issue.
